### DISCUSSION TITLE
aiolimiter running out of requests

### DISCUSSION THREAD

**Discussion by watchingdogs** [mjpieters/aiolimiter#91] [General]
I've noticed that if I set the limiter as `AsyncLimiter(180, 60)`, the code will try to send all the requests available without a limit, but then it slows down. However at that point the request speed it chooses still makes the code go over the rate limit.

My solution to the problem was to set the rate limiter to `AsyncLimiter(2.9, 1)`.
The code somehow still ran out of requests available in the one minute frame when using ` 3 ` requests as a parameter. So by sacraficing a bit of performance I was able to get the code run fine. The issue here could have been cause by the server wrongly handling requests on their API.

Thanks for making this amazing library, it has been truly helpful. :heart:

**Comment by mjpieters** [comment]
Glad you find the library helpful!

Bursting is [covered in the documentation](https://aiolimiter.readthedocs.io/en/latest/#bursting) and is normal, expected functionality. 

With `AsyncLimiter(3, 1)` the limiter will really limit things to at most 3 entries per second, timed with the asyncio loop time function, which uses your system clock. If the API you are calling out to is still telling you you are doing more than 3 requests per second something else is wrong; either the rate limiting on that server is implemented incorrectly, your computer clock is running faster than theirs, or you are making other calls outside the limited section.

There isn't really anything we can do on the aiolimiter side, however, so I'll convert this to a discussion post instead.
